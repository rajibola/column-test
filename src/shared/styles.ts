import {StyleSheet} from 'react-native';
import colors from '../constants/colors';
import {fonts} from '../constants/fonts';
import {wp, hp} from './layout';

export const ListStyles = StyleSheet.create({
  number: {
    marginTop: hp(2),
    ...fonts[14],
    opacity: 0.6,
  },
  name: {
    ...fonts[14],
    alignSelf: 'baseline',
  },
  textContainer: {
    borderBottomWidth: hp(1),
    borderBottomColor: colors.grey,
    paddingVertical: hp(8),
    flex: 1,
  },
  contactContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: wp(20),
    marginBottom: hp(15),
    flex: 1,
  },

  initials: {
    ...fonts[16],
    color: colors.white,
    textTransform: 'uppercase',
  },
  noAvatar: {
    width: wp(45),
    height: wp(45),
    backgroundColor: colors.dark,
    borderRadius: wp(45),
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: wp(10),
  },
});
export const ButtonStyles = StyleSheet.create({
  buttonText: {
    ...fonts[16],
    textTransform: 'uppercase',
  },
  container: {
    height: hp(57),
    paddingHorizontal: wp(36),
    backgroundColor: colors.light,
    position: 'absolute',
    zIndex: 10,
    bottom: hp(26),
    alignSelf: 'center',
    width: wp(342),
    borderRadius: wp(57),
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export const TextStyles = StyleSheet.create({
  RegularText: {
    ...fonts[12],
    fontFamily: 'rubik-regular',
  },

  medium: {
    ...fonts[12],
    fontFamily: 'rubik-medium',
  },
});
