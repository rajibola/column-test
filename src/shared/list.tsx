import React from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import {SharedElement} from 'react-navigation-shared-element';
import {ListItemProps} from '../types/types.d';
import {getInitails} from '../utils/helpers';
import {wp} from './layout';
import {ListStyles as styles} from './styles';
import {MediumText, RegularText} from './text';

export const ListItem = (props: ListItemProps) => {
  const {navigation, item} = props;
  return (
    <TouchableOpacity style={styles.contactContainer} onPress={navigation}>
      <SharedElement id={`item.${item.recordID}.icon`}>
        {item.hasThumbnail ? (
          <Image
            source={{uri: item.thumbnailPath}}
            width={wp(45)}
            height={wp(45)}
            style={styles.noAvatar}
          />
        ) : (
          <View style={styles.noAvatar}>
            <MediumText
              style={styles.initials}
              title={getInitails(item.givenName)}
            />
          </View>
        )}
      </SharedElement>

      <View style={styles.textContainer}>
        <SharedElement id={`item.${item.recordID}.title`}>
          <MediumText
            style={styles.name}
            title={item.givenName + ' ' + item.familyName}
          />
        </SharedElement>
        <RegularText
          style={styles.number}
          title={item?.phoneNumbers[0]?.number}
        />
      </View>
    </TouchableOpacity>
  );
};
