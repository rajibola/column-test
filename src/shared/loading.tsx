import React from 'react';
import {ActivityIndicator, View} from 'react-native';
import colors from '../constants/colors';
import {LoadingProps as Props} from '../types/types.d';
import {LoadingStyles as styles} from './styles';

export const Loading: React.FC<Props> = ({color, bgColor}) => (
  <View style={[styles.container, {backgroundColor: bgColor}]}>
    <ActivityIndicator color={color || colors.dark} size="large" />
  </View>
);
