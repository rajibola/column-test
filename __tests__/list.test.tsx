import React from 'react';
import renderer from 'react-test-renderer';
import {ListItem} from '../src/shared/list';
import {RegularText} from '../src/shared/text';

export const createTestProps = (props: any) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

describe('Button', () => {
  const Data = {
    recordID: '6b2237ee0df85980',
    emailAddresses: [
      {
        label: 'work',
        email: 'carl-jung@example.com',
      },
    ],
    familyName: 'Jung',
    givenName: 'Carl',
    phoneNumbers: [
      {
        label: 'mobile',
        number: '(555) 555-5555',
      },
    ],
    hasThumbnail: true,
    thumbnailPath: 'content://com.android.contacts/display_photo/3',
  };
  const Data2 = {
    recordID: '6b2237ee0df85980',
    emailAddresses: [
      {
        label: 'work',
        email: 'carl-jung@example.com',
      },
    ],
    familyName: 'Jung',
    givenName: 'Carl',
    phoneNumbers: [
      {
        label: 'mobile',
        number: '(555) 555-0000',
      },
    ],
    hasThumbnail: true,
    thumbnailPath: 'content://com.android.contacts/display_photo/3',
  };
  it('Button component renders correctly', () => {
    const instance = renderer.create(
      <ListItem item={Data} navigation={() => createTestProps({})} />,
    );
    expect(instance).toMatchSnapshot();
  });
  it('Check if numbers render correctly', () => {
    const instance = renderer.create(
      <ListItem item={Data} navigation={() => createTestProps({})} />,
    );
    const textInstance = instance.root.findByType(RegularText);
    expect(textInstance.props.title).toBe('(555) 555-5555');

    instance.update(
      <ListItem item={Data2} navigation={() => createTestProps({})} />,
    );
    expect(textInstance.props.title).toBe('(555) 555-0000');
  });
});
