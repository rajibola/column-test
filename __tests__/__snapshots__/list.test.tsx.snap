// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 27.5,
      "opacity": 1,
      "paddingLeft": 36,
    }
  }
>
  <Image
    height={81.5}
    source={
      Object {
        "uri": "content://com.android.contacts/display_photo/3",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#34495e",
        "borderRadius": 81.5,
        "height": 81.5,
        "justifyContent": "center",
        "marginRight": 18,
        "width": 81.5,
      }
    }
    width={81.5}
  />
  <View
    style={
      Object {
        "borderBottomColor": "#95a5a6",
        "borderBottomWidth": 2,
        "flex": 1,
        "paddingVertical": 14.5,
      }
    }
  >
    <View
      collapsable={false}
    >
      <Text
        style={
          Array [
            Object {
              "color": "#34495e",
              "fontFamily": "rubik-medium",
              "fontSize": 22,
              "lineHeight": 33,
            },
            Object {
              "alignSelf": "baseline",
              "color": "#34495e",
              "fontSize": 25.5,
              "lineHeight": 38.5,
            },
          ]
        }
      >
        Carl Jung
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#34495e",
            "fontFamily": "rubik-regular",
            "fontSize": 22,
            "lineHeight": 33,
          },
          Object {
            "color": "#34495e",
            "fontSize": 25.5,
            "lineHeight": 38.5,
            "marginTop": 7.5,
            "opacity": 0.6,
          },
        ]
      }
    >
      (555) 555-5555
    </Text>
  </View>
</View>
`;
